### ISSUE TITLE
Implementation of usage limits on services providers (OTC, Exoscale, Cyfronet) side

### ISSUE THREAD

**Issue by cgizzi** [rhea-group/hn-prototype-platform#10]
Storage and compute resources usage must be capped by service providers to avoid accidental over-usage.
[ ] OTC
[ ] Exoscale
[ ] Onedata

**Comment by eeckstei** [comment]
Within OTC Service Quota Function can be used to prevent the overuse of resources.

How this works you can find here: [OTC Service Quota:](https://docs.otc.t-systems.com/faq/iaas/en-us_topic_0040259342.html?agencyId=cc4572adbb83493da7f01b8906eb6c59&region=eu-de&locale=en-us&regionId=eu-de)

**Comment by retrack** [comment]
Same on Exoscale, we have limits which can be viewed online per organization: 
![image](https://cloud.githubusercontent.com/assets/1295273/25613336/2588c83e-2f2e-11e7-94bf-b417664fc220.png)

Adjusting limits is handled via this interface.

**Comment by cgizzi** [comment]
@eeckstei @retrack We must be sure that the Buyers will not be enabled to use more than the following resources in total for each Test Phase (TP):
TP2: 100 cores+10TB
TP3 2000 cores+200TB
TP4: 3500 cores+350TB
In addition to that they must not be enabled to buy additonal services direclty on OTC and Exoscale portals.  @retrack Is that possible for Exoscale? 
@eeckstei Is that possible on OTC? (by the way I do not see commands to change quotas at the link you sent out, just a telephone number are you going to answer ;-))

**Comment by eeckstei** [comment]
@cgizzi: Yes, it is possible to restrict resources at OTC. It must be requested via HelpCenter from the Tenant owner (via Mail or via Telephone). 
The open questions for me are: 
@zhenggenqiang: 3 questions for you:
Is it possible to set some Service Quotas be set to 0?
Is OBS restricted via Disk Service Quota also and when no how we can restrict OBS?
Are HPCI and HPCII vCPUS are restriced via vCPU-Service Quotas also and when no how we can restrict them?

**Comment by zhenggenqiang** [comment]
OBS currently doesn't support quota feature, but each user can only create 100 buckets as max. In addition, if we want to support quota feature, which parameters are needed to limit or restrict.

**Comment by zhenggenqiang** [comment]
@eeckstei 
1: This question I have to check with my colleague and give the answer ASAP
2: OBS currently doesn't support quota feature, but each user can only create 100 buckets as max. In addition, if we want to support quota feature, which parameters are needed to restrict.
3: HPCI and HPCII vCPUS don't have independent quota restrictions, all types of ECS quota are uniformly restricted.

**Comment by retrack** [comment]
@zhenggenqiang, @cgizzi On Exoscale, these limits can be enforced on a per account basis. However since multiple organization have already been created for HNsciCloud it is not possible to SUM the usage of all those organizations and enforce the overall maximum limits.

Please instruct on per organization limits or group all usages under one organization only.

**Comment by mebster** [comment]
We will have to be pragmatic here.  I suggest we limit the quotas on both Nuvla and OTC/Exoscale levels and monitor closely their usage level.  Once we have better level of quota granularity in Nuvla, it'll be easier to enforce these quotas and raise alarms when quotas that Nuvla is not enforcing (e.g. OBS) are reaching the quota limit.

**Comment by mebster** [comment]
Is there more we can do here?

**Comment by cgizzi** [comment]
@mebster I think yes. T-Systems/Exoscale have to define the quotas corresponding to the resource amounts I have reported above (per TP) and put the corresponding limits on their portals. This require further analysis of the BG needs and maybe further iterations with the BG.

**Comment by cgizzi** [comment]
Usage monitoring done manually for the first period.
